vulnerability/events_creator_test.go

```diff
@@ -117,11 +119,33 @@ var combinedCvss = types.DetectedVulnerability{
 	},
 }
 
-func TestEventsCreator_CreateEvents_Success(t *testing.T) {
-	provider := &dataprovider.MockCommonDataProvider{}
-	provider.EXPECT().EnrichEvent(mock.Anything, mock.Anything).Return(nil).Times(55)
+type EventsCreatorTestSuite struct {
+	suite.Suite
+	cdp     *MockGlobalDataProvider
+	bdp     *dataprovider.MockCommonDataProvider
+	creator EventsCreator
+}
+
+func TestSuite(t *testing.T) {
+	s := new(EventsCreatorTestSuite)
+	suite.Run(t, s)
+}
+
+func (s *EventsCreatorTestSuite) SetupTest() {
+	s.cdp = &MockGlobalDataProvider{}
+	s.bdp = &dataprovider.MockCommonDataProvider{}
```

Comment:
FYI, creating mocks with the constructor will also assert that all expectations met.
For example in that case you can use `NewMockGlobalDataProvider`.